frontend/packages/kubevirt-plugin/src/components/LabelsList/labels-list.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import {
+  Grid,
+  GridItem,
+  Text,
+  TextVariants,
+  Button,
+  Split,
+  SplitItem,
+} from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { ExternalLink, resourcePath } from '@console/internal/components/utils';
+import { LabelRow } from './LabelRow/label-row';
+import { ADD_LABEL, LABEL_KEY, LABEL_VALUE } from './consts';
+import { IDLabel } from '../../hooks/use-id-labels';
+import './labels-list.scss';
+
+export const LabelsList = <T extends IDLabel>({
+  labels,
+  onLabelAdd,
```

Comment:
it is much safer - also type-wise for the caller to pass a function like this

`onLabelAdd = () => add({ key: '', value: '' })`

plus no need for the id - as it will be supllied

Reply:
can we make it generic and not pass newLabel?